contracts/DomainRegistryV2.sol

```diff
@@ -0,0 +1,310 @@
+        return _getDomainRegistryStorage().domainHolderReward;
+    }
+
+    /// @notice Get the total reward amount for a domain
+    function getDomainRewardAmount(string memory domainName) external view returns (uint256) {
+        return _getDomainRegistryStorage().domainToReward[domainName];
+    }
+
+    /// @notice Get the total rewards amount for all domains
+    function getTotalDomainRewardAmount() external view returns (uint256) {
+        return _getDomainRegistryStorage().totalDomainRewardsAmount;
+    }
+
+    /**
+    * @notice Registers a new top-level domain
+    * @dev Emits a `DomainRegistered` event upon success
+    * @param _domain The domain name to register
+    */
+    function registerDomain(string calldata _domain) payable external availableDomain(_domain) {
+        DomainRegistryStorage storage $ = _getDomainRegistryStorage();
+
+        if (msg.value != $.registrationFee) revert IncorrectRegistrationFee($.registrationFee);
+        if ($.domainToHolder[_domain] != address(0x0)) revert DomainAlreadyRegistered(_domain);
+
+        $.domainToHolder[_domain] = payable(msg.sender);
+
+        emit DomainRegistered(_domain, msg.sender);
+
+        _applyRewardToParentDomainHolder(_domain);
+    }
+
+    /**
+    * @notice Changes the registration fee for domain registration
+    * @dev Emits a `RegistrationFeeChanged` event upon success
+    * @param _newFee The new fee for registering a domain
+    */
+    function changeRegistrationFee(uint256 _newFee) external onlyOwner {
+        DomainRegistryStorage storage $ = _getDomainRegistryStorage();
+
+        if (_newFee <= 0) revert RegistrationFeeMustBeGreaterThanZero();
+        if (_newFee == $.registrationFee) revert NewRegistrationFeeMustDifferFromCurrent();
+
+        $.registrationFee = _newFee;
+        emit RegistrationFeeChanged(_newFee);
+    }
+
+    /// @notice Change the reward amount for domain holders
+    function  changeDomainHolderReward(uint256 _newRewardAmount) external onlyOwner {
+        DomainRegistryStorage storage $ = _getDomainRegistryStorage();
+
+        if (_newRewardAmount <= 0) revert DomainHolderRewardMustBeGreaterThanZero();
+        if (_newRewardAmount == $.domainHolderReward) revert NewDomainHolderRewardMustDifferFromCurrent();
+
+        $.domainHolderReward = _newRewardAmount;
+        emit DomainHolderRewardChanged(_newRewardAmount);
+    }
+
+    /**
+    * @notice Withdraws the collected registration fees to the owner's address
+    * @dev Can only be called by the contract owner
+    */
+    function withdrawFees() external onlyOwner {
+        uint256 feeAmountToWithdraw = address(this).balance - _getDomainRegistryStorage().totalDomainRewardsAmount;
+        bool success = _transferEtherTo(payable(owner()), feeAmountToWithdraw);
+
+        if (!success) revert FailedToWithdrawFees();
+    }
+
+    /// @notice Withdraw rewards from a domain
+    function withdrawRewardForDomain(string memory _domain) external onlyOwner {
```

Comment:
// and possible 
// it is not goo solution withdraw for owner only
// should be possible to withdraw for any address
// if this address is the owner of domain